text/0000-enum-variant-types.md

```diff
@@ -0,0 +1,203 @@
+- Feature Name: `enum_variant_types`
+- Start Date: 03-11-2018
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Enum variants are considered types in their own rights. This allows them to be irrefutably matched
```

Comment:
```suggestion
Consider enum variants types in their own rights. This allows them to be irrefutably matched
```
The way you've worded it doesn't make it clear that you are changing something and could be interpreted as a description of the status quo.